csharp/src/Ice/Internal/Instance.cs

```diff
@@ -43,9 +43,27 @@ public void updateThreadObservers()
         private Instance _instance;
     }
 
-    public bool destroyed()
+    internal Task shutdownCompleted
```

Comment:
This PR is purely about providing a nicer API since the implementation waits synchronously.